### PR TITLE
Change basic_statistics API for 2025.0

### PR DESCRIPTION
**Pull request by icfaust**
### Description
---
Address promised warning in code set for 2025.0 rollout. This was originally implemented in #1644 
Necessary for enabling oneapi-src/oneDAL#2889

Checklist to comply with before moving PR from draft:

**PR completeness and readability**

- [ ] I have reviewed my changes thoroughly before submitting this pull request.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [ ] I have added a respective label(s) to PR if I have a permission for that.  
- [ ] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [ ] The unit tests pass successfully.
- [ ] I have run it locally and tested the changes extensively.

**Performance**

- [ ] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [ ] I have provided justification why performance has changed or why changes are not expected.

### CODE REVIEW

**Comment by icfaust** [comment]
> https://github.com/intel/scikit-learn-intelex/blob/c214145e26dfc0e19aadaac5648c7284f63bacc7/sklearnex/basic_statistics/basic_statistics.py#L110
> 
> 
> it is also necessary to remove from here

Complete!

**Review by olegkkruglov — APPROVED** [review]
LGTM, thanks for the addressing. assuming green CI before merge.

**Review by samir-nasibli — APPROVED** [review]

**Comment by icfaust** [comment]
Other private CI failed due to abort: http://intel-ci.intel.com/ef6ab79f-f0c2-f1ee-8460-a4bf010d0e2e
